content/posts/Jaydip_Dey_Phase_1_Report.md

```diff
@@ -0,0 +1,348 @@
+  - `status`: Fetch questions based on user status (`unattempted`, `attempted`, `solved`).
+  - `search`: Search questions by `heading` or `statement`.
+  
+- **Private Methods**:
+  - `authorize_moderator`: Ensures the user is a moderator.
+  - `set_question`: Fetches a question by ID.
+  - `question_params`: Permits parameters for `Question` creation and updates.
+
+##### QuestionSubmission Controller
+- **Functions:**
+  - `post_submission`: Handles user submission of responses.
+  - `show_submission`: Retrieves a specific submission from the user's history.
+
+##### User Controller Enhancements
+
+- **New Fields:**
+  - `submission_history` (JSONB array of submissions)
+  - `public` (Boolean to toggle dashboard privacy)
+  - `question_bank_moderator` (Boolean to manage moderator access)
+
+- **New Methods:**
+  - `submission_status`: Retrieves the submission history for a user.
+  - `toggle_dashboard_privacy`: Toggles the user's dashboard privacy setting.
+
+
+For more detailed information, please refer to [this](https://medium.com/@jaydipdey2807/week-1-3-at-circuitverse-google-summer-of-code-2024-9accce10639d) blog post
+
+---
+
+### Manage question moderators
+
+
+In the fourth week, the focus was on enhancing the admin dashboard to allow admins to manage moderators who can create questions on the UI. A new button was added to the admin's dashboard to facilitate the addition and removal of moderators.
+
+![add-moderator](/images/Jaydip_GSoC24/add_moderator_button.webp)
+
+#### UI Changes
+
+- **Dashboard Button:**
+  - Added a new button to the admin's dashboard to manage moderators.
+  - Modified the `app/views/users/circuitverse/index.html.erb` file:
+
+    ```erb
+    <% if current_user.admin? %>
+      <button type="button" class="btn btn-secondary ml-0" data-toggle="modal" data-target="#addModeratorsModal"><%= t("users.circuitverse.index.add_moderator") %></button>
+    <% end %>
+
+    <%= render partial: "add_moderators_modal" %>
+    ```
+
+- **Modal for Managing Moderators:**
+  - When the button is clicked, a modal opens where admins can enter email addresses of users to be added as moderators or remove existing moderators.
+
+![manage-moderator-1](/images/Jaydip_GSoC24/manage_mod_1.webp)
+
+#### Controller Functions
+
+- **Add Moderators:**
+  - The `add_moderators` function processes the email addresses entered by the admin and updates the `question_bank_moderator` field for the corresponding users.
+
+    ```
+    def add_moderators
+      emails = params[:emails].split(',').map(&:strip)
+      users = User.where(email: emails)
+      
+      users.each do |user|
+        user.update(question_bank_moderator: true)
+      end
+      
+      if users.all? { |user| user.errors.empty? }
+        redirect_back(fallback_location: root_path)
+      else
+        render json: { errors: users.map(&:errors).flat_map(&:full_messages) }, status: :unprocessable_entity
+      end
+    end
+    ```
```

Comment:
Avoid adding code snippets to this blog, until unless it's too important.
ig All sort of these things are present in your personal blog. Adding link to that blog or code snippet will be better way.

Reply:
Same for all the code snippets in the blog @jaydip1235

Reply:
You may add gif or video of the feature in the blog also to showcase a ppreview of the feature.